### PR TITLE
Update MNNLoader to generic

### PR DESCRIPTION
**Pull request by scorpioluck20**

### CODE REVIEW

File: src/MNNLoader.cpp [hunk 1]
```diff
@@ -1,132 +1,23 @@
...
+    }
 
-    void MNNLoader::initialize()
+    std::shared_ptr<void> MNNLoader::LoadFile(std::string filename)
     {
+        m_log_id = filename.data();
+        m_mnn_file_path = filename.data();
         // Initialize the interpreter
         m_mnn_interpreter = std::shared_ptr<MNN::Interpreter>(
                 MNN::Interpreter::createFromFile(m_mnn_file_path));
```
Comment (Super-Genius) [inline src/MNNLoader.cpp:19]: This shouldn't be just a call to the MNN::Interpreter to load the file how it sees fit.

The biggest issue, is that you should add Async Load/Parse and Save Functions.

Then this MNNLoader should be an async Saver from a void pointer not a loader directly from the file.

Hopefully the MNN::Interpreter, really should act like this.

We need to be able to stream from ipfs: or ftp: or file:  or at least:

LoadAsync("test.mnn", [](void *loadout) { ParseAsync(loadout, [](void *pout) { SaveAsync("mnn", pout) } ) } );

- or -

Pseudo code

shared_ptr<void> LoadAsync(std::string filePath, std::string suffix, shared_ptr<void> *callback());

void ParseFinishedCallback(std::string suffix, shared_ptr<void> asyncOut) {
  SaveAsync(filePath, suffix, asyncOut);  <---- The MNN::Interpreter, will be called to save the data into the interpreter.
}

void LoadFinishedCallback(std::string filePath, std::string suffix, std::string<void> loadout) {
      ParseAsync(filePath, suffix, loadout, ParseFinishedCallback); <---- MNN parse could do nothing, if it doesn't convert from XML/JSON to binary for instance.
}

LoadAsync("file://test.mnn", "mnn", LoadFinishedCallBack);  <--- calls the regular async file load in C++ standard libraray

**Review by Super-Genius — CHANGES_REQUESTED** [review]
MNNLoader should be MNNSaver, not Loader.  See other comments

**Comment by scorpioluck20** [comment]
> MNNLoader should be MNNSaver, not Loader. See other comments

We should create MNNSaver for save action.
If we rename MNNLoader to MNNSaver, I think we should delete IPFSLoader too.
We should have CommonLoader for almost of type prefix.
CommonLoader will load all contents to the buffer, and FileParser/FileSaver will be working on this buffer.
The special prefix loader (Need extra action in the loading step), will be defined and registered after.
What's do you thing?

File: src/MNNLoader.cpp (hunk 1 above)
Comment (scorpioluck20) [reply src/MNNLoader.cpp]: Updated!

**Comment by scorpioluck20** [comment]
Hi @Super-Genius ,
I suggest merging this PR and close #1 . Then we can continue with Async Loader/Parsing/Saving task in #3 
What do you think?

**Pull request closed without merging**